astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -22,49 +22,34 @@
 
 @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, CIRS, AltAz)
 def cirs_to_altaz(cirs_coo, altaz_frame):
-    if np.any(cirs_coo.obstime != altaz_frame.obstime):
-        # the only frame attribute for the current CIRS is the obstime, but this
-        # would need to be updated if a future change allowed specifying an
-        # Earth location algorithm or something
-        cirs_coo = cirs_coo.transform_to(CIRS(obstime=altaz_frame.obstime))
-
-    # we use the same obstime everywhere now that we know they're the same
-    obstime = cirs_coo.obstime
+    if (np.any(altaz_frame.location != cirs_coo.location) or
```

Comment:
No it does not, because the default for AltAz is `None`, but the default for CIRS is `EarthLocation(0, 0, 0)`

Reply:
Ok, but the error message will change now? What is the result of:

```
from astropy.coordinates import SkyCoord, AltAz

SkyCoord(alt='70d', az='0d', frame=AltAz()).transform_to(AltAz())
```

Currently it is `"NoneType" object has no attribute "to_geodetic"`

Reply:
This PR does not change that...

Reply:
Indeed - in fact I'm not sure how one would change that. An AltAz frame without a location is undefined IMO, and the issue is with the fact that you are allowed to create one. 

Rather than fix the error message in the snippet above I would prefer the code died when attempting to create the AltAz frame, but the habit of defining AltAz frames like this, and picking up the location for the AltAz from the `from_coo` is in the wild so we are stuck with it I think

Reply:
Going to call this a separate issue - though one we should fix, if likely at a different level!

Reply:
See https://github.com/astropy/astropy/issues/6777